lib/Controller/ViewController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace OCA\Tables\Controller;
+
+use OCA\Tables\AppInfo\Application;
+use OCA\Tables\Db\Table;
+use OCA\Tables\Db\ViewMapper;
+use OCA\Tables\Errors\InternalError;
+use OCA\Tables\Errors\NotFoundError;
+use OCA\Tables\Errors\PermissionError;
+use OCA\Tables\Service\TableService;
+use OCA\Tables\Service\ViewService;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\IRequest;
+use Psr\Log\LoggerInterface;
+
+class ViewController extends Controller {
+	private ViewService $service;
+
+	private ViewMapper $mapper;
+
+	private TableService $tableService;
+
+	private string $userId;
+
+	protected LoggerInterface $logger;
+
+	use Errors;
+
+
+	public function __construct(IRequest     $request,
+		ViewService $service,
+		ViewMapper $mapper,
+		LoggerInterface $logger,
+		TableService $tableService,
+		string $userId) {
+		parent::__construct(Application::APP_ID, $request);
+		$this->service = $service;
+		$this->mapper = $mapper;
+		$this->tableService = $tableService;
+		$this->userId = $userId;
+		$this->logger = $logger;
+	}
+
+
+	/**
+	 * @NoAdminRequired
+	 */
+	public function index(int $tableId): DataResponse {
+		return $this->handleError(function () use ($tableId) {
+			return $this->service->findAll($this->getTable($tableId), $this->userId);
+		});
+	}
+
+	/**
+	 * @NoAdminRequired
+	 */
+	public function indexSharedWithMe(): DataResponse {
+		return $this->handleError(function () {
+			return $this->service->findSharedViewsWithMe($this->userId);
+		});
+	}
+
+	/**
+	 * @NoAdminRequired
+	 */
+	public function show(int $id): DataResponse {
+		return $this->handleError(function () use ($id) {
+			return $this->service->find($id);
+		});
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function create(int $tableId, string $title, string $emoji): DataResponse {
```

Comment:
https://github.com/nextcloud/tables/pull/426/files#diff-6fea2d7bc8f27b4b6b6ad26e651fa8a71fee8ee025239d3ee37cbda593b36557R207 has a null-able emoji, so maybe this should also be the case here?